lms/static/scripts/frontend_apps/components/test/GroupConfigSelector-test.js

```diff
@@ -193,22 +197,46 @@ describe('GroupConfigSelector', () => {
     assert.equal(options.length, fakeGroupSets.length);
   });
 
-  it('hides authorization prompt if checkbox is de-selected', async () => {
-    fakeAPICall
-      .withArgs(groupSetsAPIRequest)
-      .rejects(new Error('Authorization failed'));
+  it('shows errors in a modal with a retry button', async () => {
+    fakeIsAuthorizationError.returns(false);
+    fakeAPICall.withArgs(groupSetsAPIRequest).rejects(new Error('Some error'));
+    const wrapper = createComponent({
+      groupConfig: { useGroupSet: true, groupSet: null },
+    });
+
+    const authModal = await waitForElement(wrapper, 'ErrorModal');
+
+    fakeAPICall.withArgs(groupSetsAPIRequest).resolves(fakeGroupSets);
+    authModal.prop('onRetry')();
+
+    const options = await waitForElement(
+      wrapper,
+      'option[data-testid="groupset-option"]'
+    );
+    assert.equal(options.length, fakeGroupSets.length);
+  });
+
+  it('unchecks checkbox if user cancels out of error modal', async () => {
+    fakeIsAuthorizationError.returns(false);
+    fakeAPICall.withArgs(groupSetsAPIRequest).rejects(new Error('Some error'));
+    const onChangeGroupConfig = sinon.stub();
+
     const wrapper = createComponent({
       groupConfig: { useGroupSet: true, groupSet: null },
+      onChangeGroupConfig,
     });
 
-    await waitForElement(wrapper, 'AuthButton');
+    const authModal = await waitForElement(wrapper, 'ErrorModal');
 
-    // De-activate group sets. In the actual app this would be done by the user
-    // toggling the checkbox. The checkbox state lives in the parent component
-    // though.
-    wrapper.setProps({ groupConfig: { useGroupSet: false, groupSet: null } });
+    authModal.prop('onCancel')();
```

Comment:
This should be wrapped in `act`. To recap: if a callback that might schedule state updates or effects is being invoked directly rather than by Enzyme (eg. via `someWrapper.simulate(event)`) then it should be wrapped in `act`.